NPE with Codelens in empty XSD file

Create an empty XSD file, activate Codelens and type a space:

```
Caused by: java.lang.NullPointerException
	at org.eclipse.lemminx.extensions.xsd.utils.XSDUtils.getTargetAttrs(XSDUtils.java:305)
	at org.eclipse.lemminx.extensions.xsd.utils.XSDUtils.searchXSOriginAttributes(XSDUtils.java:258)
	at org.eclipse.lemminx.extensions.xsd.participants.XSDCodeLensParticipant.doCodeLens(XSDCodeLensParticipant.java:50)
	at org.eclipse.lemminx.services.XMLCodeLens.getCodelens(XMLCodeLens.java:42)
	at org.eclipse.lemminx.services.XMLLanguageService.getCodeLens(XMLLanguageService.java:244)
	at org.eclipse.lemminx.XMLTextDocumentService.lambda$codeLens$17(XMLTextDocumentService.java:377)
	at java.base/java.util.concurrent.CompletableFuture.biApply(CompletableFuture.java:1236)
```

--- a/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xsd/utils/XSDUtils.java
+++ b/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xsd/utils/XSDUtils.java
@@ -288,6 +288,9 @@ public class XSDUtils {
 	 * @return the referenced attributes list from the given referenced node.
 	 */
 	private static List<DOMAttr> getTargetAttrs(DOMNode referencedNode) {
+		if (referencedNode == null) {
+			return Collections.emptyList();
+		}
 		List<DOMAttr> referencedNodes = new ArrayList<>();
 		Document document = referencedNode.getOwnerDocument();
 		switch (referencedNode.getNodeType()) {
@@ -302,7 +305,11 @@ public class XSDUtils {
 			// The referenced node is the DOM document, collect all attributes
 			// xs:complexType/@name, xs:simpleType/@name, xs:element/@name, xs:group/@name
 			// which can be referenced
-			NodeList nodes = document.getDocumentElement().getChildNodes();
+			Element documentElement = document.getDocumentElement();
+			if (documentElement == null) {
+				break;
+			}
+			NodeList nodes = documentElement.getChildNodes();
 			for (int i = 0; i < nodes.getLength(); i++) {
 				Node n = nodes.item(i);
 				if (n.getNodeType() == Node.ELEMENT_NODE) {

--- a/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/xsd/XSDCodeLensExtensionsTest.java
+++ b/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/xsd/XSDCodeLensExtensionsTest.java
@@ -75,4 +75,16 @@ public class XSDCodeLensExtensionsTest {
 				cl(r(15, 16, 15, 35), "1 reference", SHOW_REFERENCES));
 	}
 
+	@Test
+	public void codeLensEmptyDocument() throws BadLocationException {
+		String xml = "";
+		XMLAssert.testCodeLensFor(xml);
+	}
+
+	@Test
+	public void codeLensSpace() throws BadLocationException {
+		String xml = " ";
+		XMLAssert.testCodeLensFor(xml);
+	}
+
 }

